newlib/libc/sys/vita/fs.c

```diff
@@ -592,3 +595,71 @@ int fchdir(int fd)
 	}
 	return 0;
 }
+
+int statvfs(const char *__path, struct statvfs *__buf)
+{
+	if (!__buf || !__path)
+	{
+		errno = EFAULT;
```

Comment:
an error should be one of `EIO`, `EINTR`, `EOVERFLOW`, `EACCESS`, `ELOOP`, `ENAMETOOLONG`, `ENOENT`, `ENOTDIR`

https://pubs.opengroup.org/onlinepubs/9699919799/functions/statvfs.html

Reply:
No, this is what linux returns if path is NULL. (and it actually segfaults if buf is NULL)

Reply:
i mean, in spec, `statvfs` only can set one of these errors to errno.

Reply:
Yes, but afaik ENOSYS/EFAULT are always valid errors

Reply:
i saw some function definied an error cases of the ENOSYS/EFAULT. (but i don't remember that)
instead of it, i found [general purpose of error types](https://pubs.opengroup.org/onlinepubs/9699919799/functions/V2_chap02.html#tag_15_03)

```
[ENOSYS]
    Functionality not supported. An attempt was made to use optional functionality that is not supported in this implementation.

[EFAULT]
    Bad address. The system detected an invalid address in attempting to use an argument of a call. The reliable detection of this error cannot be guaranteed, and when not detected may result in the generation of a signal, indicating an address violation, which is sent to the process.
```

so... `EFAULT` is matched error type.

----
additionally,

probably you already know, opengroup just defined posix standard, but it's smallset of real os,
as this reason, `statvfs` of linux manpage is defined ENOSYS case and EFAULT case.
and also, freebsd is same as linux, so i'm changing of mind, it's ok